Web3 js installation error

I want to install webjs so I run the below command but it is showing an error on Ubuntu OS.      
    
>$ sudo npm install -g web3
    
    npm ERR! path git
    npm ERR! code ENOENT
    npm ERR! errno ENOENT
    npm ERR! syscall spawn git
    npm ERR! enoent Error while executing:
    npm ERR! enoent undefined ls-remote -h -t ssh://git@github.com/web3-js/WebSocket-Node.git
    npm ERR! enoent 
    npm ERR! enoent 
    npm ERR! enoent spawn git ENOENT
    npm ERR! enoent This is related to npm not being able to find a file.
    npm ERR! enoent 

    npm ERR! A complete log of this run can be found in:
    npm ERR!     /home/nithin/.npm/_logs/2019-09-18T09_41_14_857Z-debug.log

> I want to install webjs so I run the below command but it is showing an error on Ubuntu OS.
> 
> > $ sudo npm install -g web3
> 
> ```
> npm ERR! path git
> npm ERR! code ENOENT
> npm ERR! errno ENOENT
> npm ERR! syscall spawn git
> npm ERR! enoent Error while executing:
> npm ERR! enoent undefined ls-remote -h -t ssh://git@github.com/web3-js/WebSocket-Node.git
> npm ERR! enoent 
> npm ERR! enoent 
> npm ERR! enoent spawn git ENOENT
> npm ERR! enoent This is related to npm not being able to find a file.
> npm ERR! enoent 
> 
> npm ERR! A complete log of this run can be found in:
> npm ERR!     /home/nithin/.npm/_logs/2019-09-18T09_41_14_857Z-debug.log
> ```

Did you issue get resolve
i am getting same error need your help????

> > I want to install webjs so I run the below command but it is showing an error on Ubuntu OS.
> > > $ sudo npm install -g web3
> > 
> > 
> > ```
> > npm ERR! path git
> > npm ERR! code ENOENT
> > npm ERR! errno ENOENT
> > npm ERR! syscall spawn git
> > npm ERR! enoent Error while executing:
> > npm ERR! enoent undefined ls-remote -h -t ssh://git@github.com/web3-js/WebSocket-Node.git
> > npm ERR! enoent 
> > npm ERR! enoent 
> > npm ERR! enoent spawn git ENOENT
> > npm ERR! enoent This is related to npm not being able to find a file.
> > npm ERR! enoent 
> > 
> > npm ERR! A complete log of this run can be found in:
> > npm ERR!     /home/nithin/.npm/_logs/2019-09-18T09_41_14_857Z-debug.log
> > ```
> 
> Did you issue get resolve
> i am getting same error need your help????

Hi, yes we did. That version had git as dependency

In order to resolve this issue we installed git for windows from here 
https://git-scm.com/download/win or here https://gitforwindows.org/

Thank you Bro.....

On Sun, 27 Oct 2019 at 10:19, Stanislav Simunec <notifications@github.com>
wrote:

> I want to install webjs so I run the below command but it is showing an
> error on Ubuntu OS.
>
> $ sudo npm install -g web3
>
> npm ERR! path git
> npm ERR! code ENOENT
> npm ERR! errno ENOENT
> npm ERR! syscall spawn git
> npm ERR! enoent Error while executing:
> npm ERR! enoent undefined ls-remote -h -t ssh://git@github.com/web3-js/WebSocket-Node.git
> npm ERR! enoent
> npm ERR! enoent
> npm ERR! enoent spawn git ENOENT
> npm ERR! enoent This is related to npm not being able to find a file.
> npm ERR! enoent
>
> npm ERR! A complete log of this run can be found in:
> npm ERR!     /home/nithin/.npm/_logs/2019-09-18T09_41_14_857Z-debug.log
>
> Did you issue get resolve
> i am getting same error need your help????
>
> Hi, yes we did. That version had git as dependency
>
> In order to resolve this issue we installed git for windows from here
> https://git-scm.com/download/win or here https://gitforwindows.org/
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/ethereum/web3.js/issues/3071?email_source=notifications&email_token=ANKHXCIRG3WWHDE6GCTUJJDQQVMITA5CNFSM4IX542LKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOECK26LA#issuecomment-546680620>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANKHXCJMF2N2DPMUEUV2AETQQVMITANCNFSM4IX542LA>
> .
>

--- a/packages/web3-eth-accounts/package.json
+++ b/packages/web3-eth-accounts/package.json
@@ -18,7 +18,7 @@
         "eth-lib": "0.2.7",
         "ethereumjs-common": "^1.3.2",
         "ethereumjs-tx": "^2.1.1",
-        "@web3-js/scrypt-shim": "1.0.0",
+        "@web3-js/scrypt-shim": "^0.1.0",
         "underscore": "1.9.1",
         "uuid": "3.3.2",
         "web3-core": "1.2.2",

--- a/packages/web3-eth-accounts/package.json
+++ b/packages/web3-eth-accounts/package.json
@@ -18,7 +18,7 @@
         "eth-lib": "0.2.7",
         "ethereumjs-common": "^1.3.2",
         "ethereumjs-tx": "^2.1.1",
-        "scrypt-shim": "github:web3-js/scrypt-shim",
+        "scrypt-shim": "@web3-js/scrypt-shim",
         "underscore": "1.9.1",
         "uuid": "3.3.2",
         "web3-core": "1.2.2",

--- a/packages/web3-eth-accounts/package.json
+++ b/packages/web3-eth-accounts/package.json
@@ -18,7 +18,7 @@
         "eth-lib": "0.2.7",
         "ethereumjs-common": "^1.3.2",
         "ethereumjs-tx": "^2.1.1",
-        "scrypt-shim": "@web3-js/scrypt-shim",
+        "@web3-js/scrypt-shim": "1.0.0",
         "underscore": "1.9.1",
         "uuid": "3.3.2",
         "web3-core": "1.2.2",

--- a/packages/web3-providers-ws/package.json
+++ b/packages/web3-providers-ws/package.json
@@ -15,7 +15,7 @@
   "dependencies": {
     "underscore": "1.9.1",
     "web3-core-helpers": "1.2.2",
-    "websocket": "@web3-js/WebSocket-Node"
+    "@web3-js/WebSocket-Node": "1.0.29"
   },
   "devDependencies": {
     "definitelytyped-header-parser": "^1.0.1",

--- a/packages/web3-providers-ws/package.json
+++ b/packages/web3-providers-ws/package.json
@@ -15,7 +15,7 @@
   "dependencies": {
     "underscore": "1.9.1",
     "web3-core-helpers": "1.2.2",
-    "@web3-js/WebSocket-Node": "1.0.29"
+    "@web3-js/websocket": "^1.0.29"
   },
   "devDependencies": {
     "definitelytyped-header-parser": "^1.0.1",

--- a/packages/web3-providers-ws/package.json
+++ b/packages/web3-providers-ws/package.json
@@ -15,7 +15,7 @@
   "dependencies": {
     "underscore": "1.9.1",
     "web3-core-helpers": "1.2.2",
-    "websocket": "github:web3-js/WebSocket-Node#polyfill/globalThis"
+    "websocket": "@web3-js/WebSocket-Node"
   },
   "devDependencies": {
     "definitelytyped-header-parser": "^1.0.1",

